src/gt4py/next/program_processors/runners/dace_fieldview/gtir_builtin_translators.py

```diff
@@ -346,15 +346,26 @@ def translate_broadcast_scalar(
     domain_indices = sbs.Indices([dace_gtir_utils.get_map_variable(dim) for dim, _, _ in domain])
 
     assert len(node.args) == 1
-    assert isinstance(node.args[0].type, ts.ScalarType)
     scalar_expr = _parse_fieldop_arg(
         node.args[0], sdfg, state, sdfg_builder, domain, reduce_identity=None
     )
-    assert isinstance(scalar_expr, gtir_dataflow.MemletExpr)
-    assert scalar_expr.subset == sbs.Indices.from_string("0")
-    result = gtir_dataflow.DataflowOutputEdge(
-        state, gtir_dataflow.DataExpr(scalar_expr.node, node.args[0].type)
-    )
+
+    if isinstance(node.args[0].type, ts.ScalarType):
+        assert isinstance(scalar_expr, gtir_dataflow.MemletExpr)
+        assert scalar_expr.subset == sbs.Indices.from_string("0")
+        arg_node = scalar_expr.node
+        arg_type = node.args[0].type
+    elif isinstance(node.args[0].type, ts.FieldType):
+        # zero-dimensional field
+        assert len(node.args[0].type.dims) == 0
+        assert isinstance(scalar_expr, gtir_dataflow.IteratorExpr)
+        arg_node = scalar_expr.field
+        arg_type = node.args[0].type.dtype
+    else:
+        raise ValueError(f"Unexpected argument {node.args[0]} in broadcast expression.")
```

Comment:
This should probably be a `NotImplementedError`, in case in future we are going to support broadcasting of field slice.